tests/integration/aws_s3/test_aws_s3.py

```diff
@@ -198,3 +202,35 @@ def test_using_profile(self, use_datastore_profile):
         # cleanup
         for param in params:
             os.environ.pop(param)
+
+    def test_directory(self, use_datastore_profile):
+        param = self.s3["ds"] if use_datastore_profile else self.s3["s3"]
+        parquet_dir = f"/parquets{uuid.uuid4()}"
+        parquets_url = param["bucket_path"] + parquet_dir
+        #  generate dfs
+        # Define data for the first DataFrame
+        data1 = {"Column1": [1, 2, 3], "Column2": ["A", "B", "C"]}
+
+        # Define data for the second DataFrame
+        data2 = {"Column1": [4, 5, 6], "Column2": ["X", "Y", "Z"]}
+
+        # Create the DataFrames
+        df1 = pd.DataFrame(data1)
+        df2 = pd.DataFrame(data2)
+        temp_file1 = tempfile.mktemp(suffix=".parquet")
+        temp_file2 = tempfile.mktemp(suffix=".parquet")
```

Comment:
Please use with statement, in order to delete the file after the test.

Reply:
with tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file1, \
                tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file2:

Reply:
with tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file1, \
                tempfile.NamedTemporaryFile(suffix='.parquet', delete=False) as temp_file2:

Reply:
Why to use  tempfile.NamedTemporaryFile(...) ? I need only name, not opening the file

Reply:
It is better to delete the file after the test.